### PR TITLE
SDK-1300: Yotierror

### PR DESCRIPTION
**Pull request by echarrod**
Refactor error handling - do in yotierror package

### CODE REVIEW

File: yotierror/response.go [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+}
+
+// Unwrap returns the internal error for debugging
+func (e Error) Unwrap() error {
+	return e.Err
+}
+
+// Temporary indicates this error is a temporary error
+func (e Error) Temporary() bool {
+	return e.Response.StatusCode >= 500
```
[the remark this answers is not in the document]
Comment (echarrod) [reply yotierror/response.go]: Good point, fixed in 3112e8b

File: profile/service.go [hunk 2]
```diff
@@ -50,7 +50,7 @@ func GetActivityDetails(httpClient requests.HttpClient, token, clientSdkId, apiU
 
 	response, err := requests.Execute(httpClient, request, map[int]string{404: "Profile not found"}, requests.DefaultHTTPErrorMessages)
 	if err != nil {
-		return
+		return activity, yotierror.New(err, response)
```
[the remark this answers is not in the document]
Comment (echarrod) [reply profile/service.go]: Fixed in 3112e8b

File: dynamic/service.go [hunk 3]
```diff
@@ -40,7 +41,7 @@ func CreateShareURL(httpClient requests.HttpClient, scenario *Scenario, clientSd
 
 	response, err := requests.Execute(httpClient, request, ShareURLHTTPErrorMessages, requests.DefaultHTTPErrorMessages)
 	if err != nil {
-		return
+		return share, yotierror.New(err, response)
```
[the remark this answers is not in the document]
Comment (echarrod) [reply dynamic/service.go]: Fixed in 3112e8b

**Comment by davidgrayston** [comment]
@echarrod Do we need to wrap an existing error - perhaps we can simplify this a bit like this?
https://github.com/getyoti/yoti-go-sdk/compare/SDK-1300-yotierror...SDK-1300-yotierror-suggest

(I'll close this branch if this is a no go)

File: aml/service.go [hunk 4]
```diff
@@ -40,7 +41,7 @@ func PerformCheck(httpClient requests.HttpClient, profile Profile, clientSdkId,
 
 	response, err := requests.Execute(httpClient, request, httpErrorMessages)
 	if err != nil {
-		return
+		return result, yotierror.New(response, httpErrorMessages)
```
Comment (davidgrayston) [inline aml/service.go:44]:
```suggestion
		return result, err
```

**Pull request merged**